File: Turquoise/UI/Common/TQSearchBar.m
#import "TQSearchBar.h"

@implementation TQSearchBar {
  UITextField *_tq_textField;
}

- (UITextField *)tq_textField {
  if (!_tq_textField) {
    _tq_textField = [self findTextFieldRecursivelyStartingAtView:self];
  }
  return _tq_textField;
}

- (UITextField *)findTextFieldRecursivelyStartingAtView:(UIView *)view {
  if ([view isKindOfClass:[UITextField class]]) {
    return (UITextField *)view;
  }

  for (UIView *subview in view.subviews) {
    UITextField *textField = [self findTextFieldRecursivelyStartingAtView:subview];
    if (textField) {
      return textField;
    }
  }
  return nil;
}

@end

File: Turquoise/UI/Common/TQSearchBar.swift
import Foundation
import UIKit

class TQSearchBar : UISearchBar {
  private var theTextField: UITextField?

  var tq_textField: UITextField? {
    get {
      if theTextField == nil {
        theTextField = findTextFieldRecursivelyStarting(at: self)
      }
      return theTextField
    }
  }

  private func findTextFieldRecursivelyStarting(at view: UIView) -> UITextField? {
    if view.isKind(of: UITextField.self) {
      return view as? UITextField
    }

    for subview in view.subviews {
      if let textField = self.findTextFieldRecursivelyStarting(at: subview) {
        return textField
      }
    }
    return nil
  }
}
